### PR TITLE
 Enable direct tracing the autobuild build mode by default

### PR DESCRIPTION
**Pull request by henrymercer**
Merge this once we've cut the final GHES 3.14 release.

This PR removes the feature flag for using direct tracing for the autobuild build mode, thereby enabling it for GHES.

Probably the most controversial change is removing the tests that require us to have at least one feature with a `minimumVersion` and `toolsVersion` property.  I did this because if no features have a `toolsVersion` property, we'll never hit those code paths, so it isn't that important to test them.  Having a synthetic feature is an option but might create user confusion.

### Merge / deployment checklist

- [ ] Confirm this change is backwards compatible with existing workflows.
- [ ] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [ ] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ Note that the only difference between `v2` and `v3` of the CodeQL Action is the
 
 ## [UNRELEASED]
 
+- Improve the reliability and performance of analyzing code when analyzing a compiled language with the `autobuild` [build mode](https://docs.github.com/en/code-security/code-scanning/creating-an-advanced-setup-for-code-scanning/codeql-code-scanning-for-compiled-languages#codeql-build-modes) on GitHub Enterprise Server. This feature is already available to GitHub.com users. [#2235](https://github.com/github/codeql-action/pull/2235)
 No user facing changes.
```
Comment (aeisenberg) [inline CHANGELOG.md:10]: I think this line should be deleted.

**Review by aeisenberg — COMMENTED** [review]
Looks good. I'd say it's unlikely we'll ever need a minimum version feature again since we've moved to `toolsVersion` .

File: CHANGELOG.md (hunk 1 above)
Comment (henrymercer) [reply CHANGELOG.md:10]: Thanks, this was a bad auto merge.

**Comment by henrymercer** [comment]
@angelapwen would you mind reviewing the last couple of commits so we can get this in?

**Review by angelapwen — APPROVED** [review]
Yep looks good! ✨

**Pull request merged**